publish-release.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+# Tag and publish a new release or `purescm` to npmjs.org
+#
+# Usage:
+#
+# ./publish-release.sh <major|minor|patch...>
+#
+
+set -eu
+
+if [ $# -eq 0 ]
+  then
+    echo "No arguments supplied"
+    exit 1
+fi
+
```

Comment:
We should also check that we're not publishing from a random branch:
```suggestion

# Fail if we are not on trunk
BRANCH=$(git branch --show-current)
if [ "${BRANCH}" != "master" ]; then
  echo "Please checkout master branch"
  exit 1;
fi
git pull

```

Reply:
My workflow is/was that I check out the release branch, then publish (or make a PR first, then publish). So I'm actually never in master. I thought that we can't commit straight to `master`, not sure if we have set it up as protected though.

Reply:
We can commit straight to master (because we have the admin bit) and I'd prefer it for release commits, so we don't have to go through the bureaucracy of a branch/merge just for bumping a release number.

Trunk should always be release-ready anyways, so we're all set for this kind of workflow.

Reply:
Makes sense.